tests/test_parser_steps.py

```diff
@@ -0,0 +1,600 @@
+import unittest
+from tests import common
+
+
+class TestStepsParser(unittest.TestCase):
+    """
+        Test steps gherkin parser functionality of romaine's core.
+    """
+    def setUp(self):
+        """
+            Prepare the environment for testing.
+        """
+        # We're doing a lot of long tests- don't limit the diff output length
+        self.maxDiff = None
+
+    def tearDown(self):
+        """
+            Revert changes made during testing.
+        """
+        pass
+
+    def test_getting_step_no_input_modification(self):
+        """
+            Test that getting a step doesn't change the input var.
+        """
+        # Given I have Romaine core's parser
+        parser = common.get_romaine_parser()
+
+        # When I call get_step on a list containing
+        # """[
+        #    '    # Check for empty steps',
+        #    '    ',
+        #    '    When I call get_step on an empty list',
+        # ] """
+        input_var = [
+            '    # Check for empty steps',
+            '    ',
+            '    When I call get_step on an empty list',
+        ]
+        parser.step.get_step(input_var)
+
+        # Then my input variable is not modified
+        self.assertEqual(
+            input_var,
+            [
+                '    # Check for empty steps',
+                '    ',
+                '    When I call get_step on an empty list',
+            ],
+        )
+
+    def test_getting_raw_step(self):
+        """
+            Check we can get a step with no surrounding data.
+        """
+        # Given I have Romaine core's parser
+        parser = common.get_romaine_parser()
+
+        # When I call get_step on a list containing
+        # """[
```

Comment:
Yeah, let's sort these all out properly as part of implementing the tests of itself?